src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -1649,14 +1661,18 @@ private static CompletionItemBuilder CreateFilePathCompletionBuilder(string name
                 .WithSortText(GetSortText(label, priority))
                 .Build();
 
-        private static CompletionItem CreateSymbolCompletion(Symbol symbol, Range replacementRange, string? insertText = null)
+        private static CompletionItem CreateSymbolCompletion(Symbol symbol, Range replacementRange, string? insertText = null, CompletionPriority? priority = null)
         {
             insertText ??= symbol.Name;
             var kind = GetCompletionItemKind(symbol);
-            var priority = GetCompletionPriority(symbol);
+            var completion = CompletionItemBuilder.Create(kind, insertText);
 
-            var completion = CompletionItemBuilder.Create(kind, insertText)
-                .WithSortText(GetSortText(insertText, priority));
+            priority ??= GetCompletionPriority(symbol);
+            if (priority == CompletionPriority.VeryHigh)
+            {
+                completion.Preselect();
```

Comment:
What's the user experience if preselect is set? Does it make sense to set for multiple items?

Reply:
Completion `[]` is currently preselected and the list is scrolled without preselecting any resource/module symbols. What do you think about removing all `Preselect`s and instead rely on priority sorting and the fact that VS code preselects the first one?

Reply:
Is it possible to just remove the preselection for `[]`? That wouldn't be valid in a `dependsOn` array anyway, would it?

Reply:
It does not appear so. I wonder if that is showing up because the item type of the array is `module[] | (resource | module) | resource[]). The `dependsOn` array will accept a resource array symbolic reference but not a literal/inline array according to the diagnostics. I wanted to preselect all the symbols prioritized here in order to avoid breaking any other cases where a preselect would make sense or is the expected behavior since many of the other completion item functions `GetXCompletions` are not aware of what's happening in other completion item functions. 
I will try removing all `Preselect = true` from all completions (even non resource symbols) and see if that causes any issues.